ouroboros-consensus/src/Ouroboros/Consensus/Mempool/TxSeq.hs

```diff
@@ -160,11 +180,57 @@ splitAfterTicketNo (TxSeq txs) n =
     case FingerTree.split (\m -> mMaxTicket m > n) txs of
       (l, r) -> (TxSeq l, TxSeq r)
 
+-- | \( O(\log(n)) \). Split the sequence of transactions into two parts
+-- based on the given 'TxSizeInBytes'. The first part has transactions whose
+-- summed 'TxSizeInBytes' is less than or equal to the given 'TxSizeInBytes',
+-- and the second part has the remaining transactions in the sequence.
+--
+splitAfterTxSize :: TxSeq tx -> TxSizeInBytes -> (TxSeq tx, TxSeq tx)
+splitAfterTxSize (TxSeq txs) n =
+    case FingerTree.split (\m -> mSizeBytes m > n) txs of
+      (l, r) -> (TxSeq l, TxSeq r)
+
+-- | \( O(n) \). Specification of 'splitAfterTxSize'.
+--
+-- Use 'splitAfterTxSize' as it should be faster.
+--
+-- This function is used to verify whether 'splitAfterTxSize' behaves as
+-- expected.
+splitAfterTxSizeSpec :: TxSeq tx -> TxSizeInBytes -> (TxSeq tx, TxSeq tx)
+splitAfterTxSizeSpec txseq n =
+    mapTuple fromTxTickets $ go 0 [] (txTickets txseq)
+  where
+    mapTuple :: (a -> b) -> (a, a) -> (b, b)
+    mapTuple f (x, y) = (f x, f y)
+
+    go :: TxSizeInBytes
+       -> [TxTicket tx]
+       -> [TxTicket tx]
+       -> ([TxTicket tx], [TxTicket tx])
+    go _ accTickets [] = (reverse accTickets, [])
+    go accByteSize accTickets (t:ts)
```

Comment:
I would align on `accTickets`.

Personal preference:
```haskell
go accByteSize accTickets = \case
  []   -> ..
  t:ts -> ..

Reply:
Switched to a lambda `case` with guards.